### ISSUE TITLE
Review Data Visualization Lesson

### ISSUE THREAD

**Issue by lalpert** [Techtonica/curriculum#1086]
For the data visualization lesson: https://github.com/Techtonica/curriculum/blob/master/data-visualization/data-visualization.md

Apprentices mentioned this lesson was hard to follow and get working. 

Please do the full lesson and flag any instructions that are confusing, out-of-date, or have errors. 

You can make a PR to fix it or if you don't have time, comment with a list of improvements needed.

**Comment by sdspikes** [comment]
There's a link to https://bost.ocks.org/mike/bar/ in the materials section, then the Guided Practice session uses the code from that page but doesn't mention that it's the source, and both sorta jump in without much context.  The first instruction in the guided practice is "Create a div with 'chart' class, <div class="chart"></div>, in the body of the webpage", -- which webpage?  I think you just want them to make a new/empty html document to use for this, but I was confused for a bit about whether there was a particular existing page I was supposed to be editing (e.g. from the linked tutorial) vs starting a new one.

So, I went through the link.  I don't find it particularly engaging (it doesn't ask me to think for myself at all, and gives me code without motivating why we want to do each step -- it feels more like the author wrote some code to do what they wanted, then explained it line by line rather than thinking like a beginner and helping the reader understand how to go from idea to implementation). 

I also find the structure of it kinda confusing, and hard to follow -- in the "Coding a Chart, Automatically" section, it says "starting with an empty page that contains only a div of class "chart"" but actually wants you to keep the styling as well (and load d3, which the page assumes you know how to do at the top, but never reminds you to actually load it).  It then gives a block of code without explaining it, then goes line by line using a different version of the code where it's all broken out into individual variables rather than chained.  And as mentioned, it uses an old version of d3, which leads to problems with `scale.linear` vs `scaleLinear`, as well as using old style `var`.

Recommendation: fully re-write the linked tutorial with d3 v4 in a way that is more straightforward (in particular, more clarity around what to include where, when it's on a tangent vs asking you to actually modify code, etc), and have it lead to the code that's presented in the current "Let's Make a Bar Chart" section of the lesson (right now there's extra stuff with colors added in without comment, making the code look more daunting than it needs to be, build that in to the tutorial).

The practice bit to get the number at the bottom rather than the top of the bar after the transform was useful, and helped drive home the fact that attrs can be either a value or a function.  I started looking at the pie chart thing, but don't have much more time right now so won't go into that for now.
